src/server/journal/streamer.cc

```diff
@@ -348,6 +352,7 @@ void RestoreStreamer::WriteEntry(string_view key, const PrimeValue& pk, const Pr
   // TODO: From DumpObject to till Write we tripple copy the PrimeValue. It's very inefficient and
   // will burn CPU for large values.
   Write(restore_cmd_sink.str());
+  last_write_time_ns_ = absl::GetCurrentTimeNanos();
```

Comment:
check out the flow in RdbSaver::Impl::WriteRecord
before write to socket we set the time to current time
after we write to socket returns we set the time to current time
after we finish writing the buffer to socket we set time to -1
You should implement same logic here